Reorganize packaging docs to match the workflow. (#373)
getsolus/help-center-docs: docs/packaging/creating-a-new-package.md

### Update a Fork

If you already have a fork of [getsolus/packages](https://github.com/getsolus/packages) in GitHub, log into GitHub. Make sure you're looking at the master branch. Check to see that your fork is up to date with the main repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.

## Clone the packages repo / update your clone

### Clone the repo

From within your packaging directory, clone the fork. For example `gh repo clone yourgithubaccount/packages`.
Then, change to this directory and create a new branch.

```bash
cd /home/solus-builds/packages
git switch -c your-branch
```

### Update an existing clone

If you already have a local clone, you need to bring it up to date and create a new branch. To do so run:

```bash
cd /home/solus-builds/packages
git switch master
git pull
git switch -c your-branch
```

## Create a New Package Directory

From within your packaging directory, create a new directory for the new package. Use the package name as the directory name.

> Initialize the folder as a git repository and switch to a new branch.  This will allow you to more easily separate your work from any new changes made to the package repository, which will allow you to more easily rebase any changes if needed. Then, create the required Makefile within that directory.

Reviewer: ```suggestion
Initialize a folder in the packages repository and switch to a new branch.  This will allow you to more easily separate your work from any new changes made to the packages repository, which will allow you to more easily rebase any changes if needed. Then, create the required Makefile within that directory.
```

Author: Thanks for clarifying. I've updated this section.

```bash
mkdir tree
cd tree
git init

> git switch -c my-branch

Reviewer: These two lines are not needed. The repo is setup above.

Author: Just above you recommended
"Initialize a folder in the packages repository and switch to a new branch."

So it would still need to be 
```bash
mkdir tree
cd tree
git switch -c my-branch
```

(I have removed the new branch command that was before this so the user is only doing it once)

```

> ## Link to the `Makefile` File

Author: Agreed, fixed

> Every package directory needs a file `Makefile`. To create it with the proper contents, run this:

Author: Agreed, fixed

---
title: Creating a New Package
summary: Creating a Package For Inclusion in the Repo
sidebar_position: 3
---

# Creating a New Package

This page will take you through the steps required to build a new package for the Solus repo. For this example, we will use the command line utility [tree](https://gitlab.com/OldManProgrammer/unix-tree), which is already packaged for Solus, but we will pretend like it's not.

:::note

**Please [look to see if an issue has been filed](https://github.com/getsolus/packages/issues?q=label%3A%22Package+Request%22) for the software or library you intend to package**. If there is an existing request, please add a link to it in your pull request. Ex:

```
Software request https://github.com/getsolus/packages/issues/123
```

:::

## Create a New Package Directory

From within your packaging directory, create a new directory for the new package with the package name, then create the required Makefile within that directory:

```bash
mkdir tree
cd tree
echo "include ../Makefile.common" > Makefile
```

:::note
We recommend making the name of the package directory match the name of the package.
:::

## Initialize Git Repo

> Initialize the folder as a git repository and switch to a new branch.  This will allow you to more easily separate your work from an new changes made to the package repository, which will allow you to more easily rebase any changes if needed.

Reviewer 2: `an new changes` -> `any new changes`?

Author: Good catch. Fixed.

    install -Dm00755 $workdir/tree $installdir/usr/bin/tree
```

To read more about finding and including dependencies and other parts of `package.yml` please see [Packaging Practices](packaging-practices.md).

Understanding how to translate source code into a good `package.yml` file is the heart of packaging. If you are stumped, or have questions, **ask for help in our Solus Packaging room on [Matrix](/docs/user/contributing/getting-involved#matrix-chat).**

## Build the Package

> Build the package (`tree` uses `make`, other packages may use another command):

Reviewer 3: What build system the software being packaged uses is irrelevant; to build a package it will always be `make` (until we switch to `go-task`).

Reviewer 3: Unless you mean like `make local`? Either way, it might give readers the impression that the package's build system matters.

Author: Ah, thanks for clarifying that. I had conflated the invocation in package.yml. with the command line build command. Fixor.

└── tree-2.1.1-1-1-x86_64.eopkg
```

> All these files _except_ the `*eopkg` file(s) should be included in your pull request. You will remove them after testing the package.

Reviewer 3: "them" is ambiguous here; it could mean the eopkg files, or all of the other files.
```suggestion
All these files _except_ the `*eopkg` file(s) should be included in your pull request. You will remove the `.eopkg` files after testing the package.
```

Author: Fixed

└── tree-2.1.1-1-1-x86_64.eopkg
```

All these files _except_ the `*eopkg` file(s) should be included in your pull request. You will remove them after testing the package.

Once your package has built successfully, you will need to [test it](testing-a-package).

## Commit your Changes

Check the [changes in your files](git-basics).

[Add / remove files as necessary to the commit](git-basics.md). Then, **check your branch**.

> Run `git status`. Make sure all the files you changed are staged, and that there are no untracked files. When all it well, run `git commit`.

Reviewer: The `##` are stripped by default from the commit message :sweat: 

```suggestion
Run `git status`. Make sure all the files you changed are staged, and that there are no untracked files. When all is well, run `git commit --cleanup=scissors`.
```

Author: After discussion in Matrix, I'll change the recommended template to use **bolding** instead of the ##. This avoids having to use git flags, and keeps it simpler.
